jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -287,7 +283,7 @@ def stop_buffering(self, kernel_id):
             self.log.info("Discarding %s buffered messages for %s",
                 len(msg_buffer), buffer_info['session_key'])
 
-    def shutdown_kernel(self, kernel_id, now=False):
+    async def shutdown_kernel(self, kernel_id, now=False):
```

Comment:
Hmm, will this break existing applications that rely on `shutdown_kernel()` not being async?  Here, an "application" could be an existing extension of `MappingKernelManager`.   If it overrides `shutdown_kernel()`, does some stuff, then delegates via `super`, the call will fail with `shutdown_kernel not awaited` or something of that sort.

Reply:
I guess I still don't understand very well what can be overridden. I thought this method couldn't be called from outside of jupyter_server. Even so, why was it the only non-async method? It can potentially take time, so it is bad to block on it.
What if we have an `async_shutdown_kernel`, and keep `shutdown_kernel` blocking?